File: python/57d_1557A_EzzatAndTwoSubsequences.py
def solve(a, n):
    avg = sum(a)
    k = max(a)
    avg -= k
    avg = avg / (n-1)
    return "{:.9f}".format(avg + k)

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        a = list(map(int, input().split()))
        ans.append(solve(a, n))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/57d_1557A_EzzatAndTwoSubsequences.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t, n;
    double avg, larg, temp;
    vector<double> ans;
    cin>>t;
    for(int i = 0; i < t; i++){
        cin>>n;
        avg = 0, larg = INT_MIN;
        for(int j = 0; j < n; j++){
            cin>>temp;
            avg += temp;
            larg = max(larg, temp);
        }
        avg -= larg;
        avg = avg / (n-1);
        avg += larg;
        ans.push_back(avg);
    }
    for(auto i : ans){
        cout<<fixed<<setprecision(9)<<i<<"\n";
    }
    return 0;
}